src/mono/netcore/System.Private.CoreLib/src/ILLink/ILLink.Substitutions.wasm.xml

```diff
@@ -9,5 +9,8 @@
     <type fullname="System.Runtime.CompilerServices.RuntimeFeature">
       <method signature="System.Boolean get_IsDynamicCodeCompiled()" body="stub" value="false" />
     </type>
+    <type fullname="System.Runtime.Serialization.SerializationInfo">
```

Comment:
@eerhardt Can you comment on this from a linker perspective? Is this correct?

Reply:
I think we'll need feature switch because we want to disable more of serialization logic based on this in linker

Secondly, you need to to clean up SerializationInfo not to have static cctor for more size savings

Reply:
> I think we'll need feature switch because we want to disable more of serialization logic based on this in linker

What more logic would we want to disable? This cuts off all binary formatter serialization.

I agree this should be a full blown feature switch. That way someone can use it on other form factors, link their app, and BinaryFormatter is completely out of the app. We can still have the WASM specific substitution to shut it off always on WASM as well.

Reply:
> What more logic would we want to disable?

I was thinking about serialization attributes and deserialization ctors

Reply:
> I was thinking about serialization attributes and deserialization ctors

@marek-safar Not quite sure how to trim `SerializableAttribute` here. Presumably other serializers might want to honor it? For the deserialization ctors and _OnDeserialized_ APIs, those tend to be private anyway, so I'm hopeful the linker's standard logic will trim them.

Also, is there anything missing from this PR that would allow this to be a full-blown "feature switch"? I know we'd still need changes to the sdk repo.

Reply:
> is there anything missing from this PR that would allow this to be a full-blown "feature switch"

- Move the XML setting to https://github.com/dotnet/runtime/blob/master/src/libraries/System.Private.CoreLib/src/ILLink/ILLink.Substitutions.Shared.xml
- Add appropriate `feature` and `featurevalue` values to `method` tag
- Add setting to https://github.com/dotnet/sdk/blob/master/src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.targets#L340

Reply:
> Move the XML setting to https://github.com/dotnet/runtime/blob/master/src/libraries/System.Private.CoreLib/src/ILLink/ILLink.Substitutions.Shared.xml
Add appropriate feature and featurevalue values to method tag

I believe the intention here is to completely remove this functionality from wasm with no ability to "opt out".

If that is the case, then this entry should stay here with no feature/featurevalue tags. And the entry should be copied in `src/libraries/System.Private.CoreLib/src/ILLink/ILLink.Substitutions.Shared.xml` as well, with feature and featurevalue tags.